tests/Game.class.test.js

```diff
@@ -0,0 +1,724 @@
+        expect(state[2][1]).toBe(4);
+        expect(state[2][2]).toBe(8);
+        expect(state[2][3]).toBe(8);
+        expect(state[3][0]).toBe(16);
+        expect(state[3][1]).toBe(16);
+        expect(state[3][2]).toBe(4);
+        expect(state[3][3]).toBe(4);
+
+        expect(game2048.getScore()).toBe(64);
+      });
+
+      it('should merge cells with the same value and stack with those that are not merged down and update the score', () => {
+        const game2048 = new Game(initialStateMergingAndStacking, 0, 'playing');
+
+        game2048.moveDown();
+
+        const state = game2048.getState();
+
+        expect(state[2][0]).toBe(2);
+        expect(state[2][3]).toBe(2);
+        expect(state[3][0]).toBe(8);
+        expect(state[3][1]).toBe(4);
+        expect(state[3][2]).toBe(4);
+        expect(state[3][3]).toBe(4);
+
+        expect(game2048.getScore()).toBe(20);
+      });
+    });
+  });
+
+  describe('game status', () => {
+    it('should be a "win" after player has moved and a 2048 tile is created', () => {
+      const game2048 = new Game([
+        [0, 0, 0, 0],
+        [0, 0, 1024, 1024],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ], 0, 'playing');
+
+      game2048.moveRight();
+
+      const state = game2048.getState();
+
+      expect(game2048.getStatus()).toBe('win');
+      expect(state[1][3]).toBe(2048);
+    });
+
+    it('should be a "playing" after player has moved and board is full but new move is still possible with added tile', () => {
+      const game2048 = new Game([
+        [2, 2, 8, 16],
+        [2, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 64, 128],
+      ], 0, 'playing');
+
+      game2048.moveDown();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should be a "playing" after player has moved and board is full but move is still possible with old tiles', () => {
+      const game2048 = new Game([
+        [128, 128, 8, 16],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 64, 128],
+      ], 0, 'playing');
+
+      game2048.moveLeft();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should be a "lose" after player has moved and board is full and no more moves are possible', () => {
+      const game2048 = new Game([
+        [128, 128, 16, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 64, 128],
+      ], 0, 'playing');
+
+      game2048.moveLeft();
+
+      expect(game2048.getStatus()).toBe('lose');
+    });
+  });
+
+  describe('game reset', () => {
+    it('should reset the game score, status and set the board to the empty state after restart', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      game2048.moveLeft();
+      game2048.moveRight();
+      game2048.moveUp();
+      game2048.moveDown();
+
+      game2048.restart();
+
+      expect(game2048.getScore()).toBe(0);
+      expect(game2048.getStatus()).toBe('idle');
```

Comment:
I don't see that we described the expected reset logic in the task description
(I think it is better to keep it the same as in the original game)